internals/src/Tizen.Peripheral/Tizen.Peripheral/Gpio.cs

```diff
@@ -0,0 +1,249 @@
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Interrupt on rising only.
+        /// </summary>
+        Rising,
+
+        /// <summary>
+        /// Interrupt on falling only.
+        /// </summary>
+        Falling,
+
+        /// <summary>
+        /// Interrupt on rising and falling.
+        /// </summary>
+        Both
+    }
+
+    /// <summary>
+    /// The class allows applications to use the platform Digital Pins as Input/Output.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Gpio : IDisposable
+    {
+
+        private GpioChangePolarity _polarityType;
+        private NativeGpio.InterruptedEventCallback _interruptedEventCallback;
+
+        /// <summary>
+        /// Native handle to Gpio.
+        /// </summary>
+        private IntPtr _handle;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Invoked when Gpio pin was interrupted.
+        /// </summary>
+        public event EventHandler<PinUpdatedEventArgs> ValueChanged;
+
+        /// <summary>
+        /// Opens a GPIO pin.
+        /// </summary>
+        /// <param name="pinNumber">The GPIO pin number.</param>
+        /// <param name="mode">GPIO direction.</param>
+        public Gpio(int pinNumber, GpioPinDriveMode mode)
+        {
+            var ret = NativeGpio.Open(pinNumber, out IntPtr handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            _handle = handle;
+            PinNumber = pinNumber;
+            try
+            {
+                switch (mode)
+                {
+                    case GpioPinDriveMode.Input:
+                        ret = NativeGpio.SetEdgeMode(handle, NativeGpio.EdgeType.Both);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.In);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        SetIntteruptedCallback();
+                        break;
+                    case GpioPinDriveMode.OutputInitiallyLow:
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutLow);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        break;
+                    case GpioPinDriveMode.OutputInitiallyHigh:
+                        ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutHigh);
+                        if (ret != ErrorCode.None)
+                            throw ExceptionFactory.CreateException(ret);
+                        break;
+                }
+            } catch(Exception e) {
+                Dispose();
+                throw e;
```

Comment:
```suggestion
                throw;
```
Please check it
https://docs.microsoft.com/ko-kr/visualstudio/code-quality/ca2200?view=vs-2019